### PR TITLE
Adds editor.formatOnSave setting and biome.formatter rules

### PR DESCRIPTION
**Pull request by DyHex**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,3 +1,4 @@
 {
-  "editor.defaultFormatter": "biomejs.biome"
-}
+  "editor.defaultFormatter": "biomejs.biome",
+  "editor.formatOnSave": true
```
Comment (Shtian) [inline .vscode/settings.json:3]: Just a suggestion, but BiomeJs has a quickfix setting as well which I believe gives you autofixing of lint issues as well, not just formatting :)
```suggestion
  "editor.formatOnSave": true,
  "editor.codeActionsOnSave": {
    "quickfix.biome": true
  }
```
ref. https://biomejs.dev/reference/vscode/#fix-on-save

File: biome.json [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
+		"rules": {
+			"recommended": true
+		}
+	},
+	"formatter": {
+		"enabled": true,
+		"formatWithErrors": true,
+		"indentStyle": "tab",
+		"indentWidth": 2,
+		"lineWidth": 120,
```
Comment (Shtian) [inline biome.json:17]: Happy with 120? We were a bit back and forth between 100 and 120 :D

**Review by Shtian — APPROVED** [review]

File: .vscode/settings.json (hunk 1 above)
Comment (DyHex) [reply .vscode/settings.json:3]: Might aswell try it :)

File: biome.json (hunk 2 above)
Comment (DyHex) [reply biome.json:17]: Ah nice catch, was indeed supposed to be 100 :) Somehow managed to revert it back to 120 when extracting this into its own branch.

File: .vscode/settings.json (hunk 1 above)
Comment (DyHex) [reply .vscode/settings.json]: https://github.com/biomejs/biome/issues/1631

I'll hold off on this until they get it working with VSCode 1.83+

File: biome.json (hunk 2 above)
Comment (DyHex) [reply biome.json]: On second thought, 100 looked better on a 1080p screen, but 120 feels better on any resolution above 1080p, so think I'll leave it at 120 for now. :)

**Pull request merged**